File: Model/Source/Precipitation.vb
Imports System.ComponentModel
Imports Wells.Model.ReflectionExtension

Public Class Precipitation
    Inherits BusinessObject

    <Browsable(False)>
    Property PrecipitationDate As String

    <DisplayName("Fecha"), Browsable(True)>
    ReadOnly Property RealDate As Date
        Get
            Return Date.ParseExact(PrecipitationDate, "dd/MM/yyyy", Nothing)
        End Get
    End Property

    <DisplayName("Milímetros"), Browsable(True)>
    Property Millimeters As Double

    Shared ReadOnly Property Propeties As New Dictionary(Of String, String) From {
       {GetDisplayName(Of Precipitation)(NameOf(Precipitation.Millimeters)), NameOf(Precipitation.Millimeters)}}
End Class

File: Model/Model/src/Precipitation.cs
using System;
using System.Collections.Generic;
using System.ComponentModel;
using System.ComponentModel.DataAnnotations;
using System.Reflection;
using Wells.BaseModel.Models;

namespace Wells.Model
{
    public class Precipitation : BusinessObject
    {

        public Precipitation() { }
        
        #region Properties
        [Required, DisplayName("Fecha"), Browsable(true)]
        public DateTime PrecipitationDate { get; set; }

        [Required, DisplayName("Milímetros"), Browsable(true)]
        public double Millimeters { get; set; }
        #endregion

        [Browsable(false)]
        public static Dictionary<string, PropertyInfo> Properties
        {
            get
            {
                return GetBrowsableProperties(typeof(Precipitation));
            }
        }

        [Browsable(false)]
        public static Dictionary<string, PropertyInfo> DoubleProperties
        {
            get
            {
                return GetDoubleBrowsableProperties(typeof(Precipitation));
            }
        }

        public override int CompareTo(IBusinessObject other)
        {
            return PrecipitationDate.CompareTo((other as Precipitation).PrecipitationDate);
        }

        public static string GetDisplayName(string propertyName)
        {
            return GetDisplayName(typeof(Precipitation), propertyName);
        }
    }
}